huawei-deployer/machines.yaml.template

```diff
@@ -0,0 +1,46 @@
+items:
+- apiVersion: "cluster.k8s.io/v1alpha1"
+  kind: Machine
+  metadata:
+    generateName: huawei-master-
+    labels:
+      set: master
+  spec:
+    providerConfig:
+      value:
+        apiVersion: "huaweiproviderconfig/v1alpha1"
+        kind: "HuaweiProviderConfig"
+        project: "$GCLOUD_PROJECT"
+        zone: "$ZONE"
+        machineType: "n1-standard-2"
+        os: "ubuntu-1604-lts"
+    versions:
+      kubelet: 1.9.4
+      controlPlane: 1.9.4
+      containerRuntime:
+        name: docker
+        version: 1.12.0
+    roles:
+    - Master
+- apiVersion: "cluster.k8s.io/v1alpha1"
+  kind: Machine
+  metadata:
+    generateName: huawei-node-
+    labels:
+      set: node
+  spec:
+    providerConfig:
+      value:
+        apiVersion: "huaweiproviderconfig/v1alpha1"
+        kind: "HuaweiProviderConfig"
+        project: "$GCLOUD_PROJECT"
```

Comment:
GCLOUD_PROJECT is a google cloud construct. This template needs a more accurate provider config

Reply:
Yeah, my mistake, I think we need time to provide all those template files accurately. This pr is just a frame, certainly need further more work.  :disappointed: